### ISSUE TITLE
Calendar view is empty due to unhandled movie status

### ISSUE THREAD

**Issue by brycied00d** [Radarr/Radarr#4159]
**Describe the bug**
When opening the calendar view, the calendar itself is present however all dates are empty. This is in spite of movies being listed on the left side indicating there definitely are movies for this currently displayed month. This only happens on some months and appears to be linked to specific movies that happen to have events in the displayed month; if the movie has nothing happening in a given month, then the month displays correctly.

**To Reproduce**
1. One movie I identified as "problematic" is Buffaloed. Add this movie.
2. Wait for the data to populate in Radarr
3. Open the calendar to the month of February.

**Expected behavior**
I expect to see the normal calendar view, with movies listed for each day they're released in cinemas, home releases, etc.

**Screenshots**
Calendar view:
![image](https://user-images.githubusercontent.com/435838/74609951-9bae3c80-50a3-11ea-8d6d-88123463cc6f.png)

**Desktop (please complete the following information):**
 - OS: Windows, Linux, Android...
 - Mono Version: 5.10.1.57
 - Browser and Version: Chrome 80 & 81 (81.0.4044.17), Firefox 72.0.2
 - Version: 0.2.0.1450 (however, by no means a recent issue. I'm only just now investigating and reporting it)

**Debug Logs**
Turn on debug logs under Settings -> General and wait for the bug to occur again. **Upload the full log file here (or another site and link it). Issues will be closed, if they do not include this!**
As this is a user-interface error, I'm including the error message and backtrace from Chrome Inspector.
```
Uncaught TypeError: Cannot read property 'toLowerCase' of undefined
    at child._getStatusLevel (CalendarView.js:181)
    at CalendarView.js:163
    at Function.forEach (lodash.underscore.js:1915)
    at child.Collection.<computed> [as each] (backbone.js:953)
    at child._setEventData (CalendarView.js:148)
    at Object.options.success (backbone.js:856)
    at fire (jquery.js:3148)
    at Object.fireWith [as resolveWith] (jquery.js:3260)
    at done (jquery.js:9314)
    at XMLHttpRequest.callback (jquery.js:9718)
_getStatusLevel @ CalendarView.js:181
(anonymous) @ CalendarView.js:163
forEach @ lodash.underscore.js:1915
Collection.<computed> @ backbone.js:953
_setEventData @ CalendarView.js:148
options.success @ backbone.js:856
fire @ jquery.js:3148
fireWith @ jquery.js:3260
done @ jquery.js:9314
callback @ jquery.js:9718
XMLHttpRequest.send (async)
send @ jquery.js:9664
ajax @ jquery.js:9215
$.ajax @ jquery.ajax.js:21
Backbone.ajax @ backbone.js:1202
Backbone.sync @ backbone.js:1185
sync @ backbone.js:630
fetch @ backbone.js:860
_getEvents @ CalendarView.js:130
_viewRender @ CalendarView.js:46
publiclyTrigger @ fullcalendar.js:11027
publiclyTrigger @ fullcalendar.js:8139
triggerRender @ fullcalendar.js:8570
onDateRender @ fullcalendar.js:8544
(anonymous) @ fullcalendar.js:8507
promise.then @ fullcalendar.js:1747
executeDateRender @ fullcalendar.js:8488
(anonymous) @ fullcalendar.js:8456
runFunc @ fullcalendar.js:1811
(anonymous) @ fullcalendar.js:1828
Promise @ fullcalendar.js:1686
addTask @ fullcalendar.js:1806
requestDateRender @ fullcalendar.js:8455
handleDate @ fullcalendar.js:8434
setDate @ fullcalendar.js:8412
renderView @ fullcalendar.js:10606
prev @ fullcalendar.js:10845
buttonClick @ fullcalendar.js:9774
(anonymous) @ fullcalendar.js:9820
dispatch @ jquery.js:4670
elemData.handle @ jquery.js:4338
```

I dug into this, setting the debugging to break on exception and this is how I identified which particular movie (or at least, the first one that is) was breaking the UI render. This particular movie (Buffaloed) has a `status` of `tba` however the `MovieModel.getStatus()` function does not have a case to handle this `status` and as a result, the function returns `null`, which the UI renderer, assuming the function returned a String, calls `null.toLower()` and throws an exception. I've attached a screenshot of the debugger including the trace and watchlisted variable.

The code responsible for handling the various `status` values, complete with a `lol`: https://github.com/Radarr/Radarr/blob/v0.2.0.1450/src/UI/Movies/MovieModel.js#L16-L37

![image](https://user-images.githubusercontent.com/435838/74610047-8f76af00-50a4-11ea-84fd-ba435eb83412.png)

**Comment by brycied00d** [comment]
PS: Here's the movie details, extracted from an API request to the calendar endpoint, which should provide some additional details:
```json
  {
    "title": "Buffaloed",
    "alternativeTitles": [],
    "secondaryYearSourceId": 0,
    "sortTitle": "buffaloed",
    "sizeOnDisk": 0,
    "status": "tba",
    "overview": "Set in the underworld of debt-collecting and follows the homegrown hustler Peg Dahl, who will do anything to escape Buffalo, NY.",
    "inCinemas": "2020-02-14T00:00:00Z",
    "images": [
      {
        "coverType": "poster",
        "url": "/MediaCover/2044/poster.jpg"
      }
    ],
    "downloaded": false,
    "year": 2020,
    "hasFile": false,
    "path": "/tank/Media/Movies/Buffaloed (2020)",
    "profileId": 4,
    "pathState": "static",
    "monitored": true,
    "minimumAvailability": "released",
    "isAvailable": false,
    "folderName": "/tank/Media/Movies/Buffaloed (2020)",
    "runtime": 0,
    "cleanTitle": "buffaloed",
    "tmdbId": 547012,
    "titleSlug": "buffaloed-547012",
    "genres": [],
    "tags": [],
    "added": "2020-02-12T03:30:43.385753Z",
    "ratings": {
      "votes": 0,
      "value": 0.0
    },
    "qualityProfileId": 4,
    "id": 2044
  },
```

I forced a data refresh (after obtaining the above) and the calendar view now renders as expected. The movie's `status` has been changed to `inCinemas` which IS a value that is properly handled by `MovieModel.getStatus()`, confirming that the issue is the UI is not able to handle movies whose `status` field is not one of the established values. Updated movie details (from API request) below.

```json
  {
    "title": "Buffaloed",
    "alternativeTitles": [],
    "secondaryYearSourceId": 0,
    "sortTitle": "buffaloed",
    "sizeOnDisk": 0,
    "status": "inCinemas",
    "overview": "Set in the underworld of debt-collecting and follows the homegrown hustler Peg Dahl, who will do anything to escape Buffalo, NY.",
    "inCinemas": "2020-02-14T08:00:00Z",
    "images": [
      {
        "coverType": "poster",
        "url": "/MediaCover/2044/poster.jpg"
      },
      {
        "coverType": "fanart",
        "url": "/MediaCover/2044/fanart.jpg"
      }
    ],
    "website": "",
    "downloaded": false,
    "year": 2019,
    "hasFile": false,
    "youTubeTrailerId": "LrQZpeMcM5g",
    "studio": "Lost City",
    "path": "/tank/Media/Movies/Buffaloed (2020)",
    "profileId": 4,
    "pathState": "static",
    "monitored": true,
    "minimumAvailability": "released",
    "isAvailable": false,
    "folderName": "/tank/Media/Movies/Buffaloed (2020)",
    "runtime": 95,
    "lastInfoSync": "2020-02-16T18:20:48.745489Z",
    "cleanTitle": "buffaloed",
    "imdbId": "tt8647310",
    "tmdbId": 547012,
    "titleSlug": "buffaloed-547012",
    "genres": [],
    "tags": [],
    "added": "2020-02-12T03:30:43.385753Z",
    "ratings": {
      "votes": 3,
      "value": 4.3
    },
    "qualityProfileId": 4,
    "id": 2044
  },
```

**Comment by ta264** [comment]
Does this occur in v3? A lot of this has changed and unlikely to get backported to v0.2.

Note V3 is not stable yet so I would not advise swapping to it yet

**Comment by brycied00d** [comment]
I have not tested it on v3, no and sadly I don't really have the resources to get that setup and find another movie that would add as "tba" (or any unhandled `status` value).

**Comment by jalmansor** [comment]
I added this to my v3 instance and it displays without issue.

**Comment by ta264** [comment]
Sounds like we can close this then

**Comment by brycied00d** [comment]
Thanks @ta264 , and thank you @jalmansor for attempting to replicate the issue in V3. However, I wouldn't be so quick to close this as I don't believe @jalmansor 's test was accurate, the data provider has already changed that movie's `status` away from "tba" to "inCinemas" (as explained in https://github.com/Radarr/Radarr/issues/4159#issuecomment-586736794), thus it wouldn't have triggered the bug even in 0.2.0.

Can Radarr's database be manually edited to set a movie's `status`? Alternatively, we'd have to come up with another movie whose `status` is "tba" or another previously-unhandled value. And of course there's a chance that by the time another individual attempts to replicate, the data provider (who is it that provides `status`, TMDB?) will have changed the value to one that does not trigger the bug.

**Comment by jalmansor** [comment]
Drat, you're right. I don't have any upcoming that are labeled `tba`, but if we can find one I'm glad to try it or make a db change if necessary.

**Comment by ta264** [comment]
Looks like jumped the gun a bit. But in any case, this is unlikely to get fixed in v0.2. There have been a lot of changes around this in v3, so if we can replicate there then we can reopen.

(Just in case you haven't realised, v0.2 is critical bug fixes only at this point)
